src/codec_aom.c

```diff
@@ -715,57 +682,95 @@ static avifResult aomCodecEncodeImage(avifCodec * codec,
                 cfg->monochrome = 1;
             }
         }
+    }
+
+    avifBool dimensionChanged = AVIF_FALSE;
+    if ((cfg->g_w != image->width) || (cfg->g_h != image->height)) {
+        cfg->g_w = image->width;
+        cfg->g_h = image->height;
+        if (codec->internal->encoderInitialized) {
+            // We are not ready for dimension change for now.
+            return AVIF_RESULT_NOT_IMPLEMENTED;
+        }
+    }
+
+    if (!codec->internal->encoderInitialized || updatedConfig) {
+        int minQuantizer = AVIF_CLAMP(encoder->minQuantizer, 0, 63);
+        int maxQuantizer = AVIF_CLAMP(encoder->maxQuantizer, 0, 63);
+        if (alpha) {
+            minQuantizer = AVIF_CLAMP(encoder->minQuantizerAlpha, 0, 63);
+            maxQuantizer = AVIF_CLAMP(encoder->maxQuantizerAlpha, 0, 63);
+        }
+        lossless = ((minQuantizer == AVIF_QUANTIZER_LOSSLESS) && (maxQuantizer == AVIF_QUANTIZER_LOSSLESS));
+        cfg->rc_min_quantizer = minQuantizer;
+        cfg->rc_max_quantizer = maxQuantizer;
 
         if (!avifProcessAOMOptionsPreInit(codec, alpha, cfg)) {
             return AVIF_RESULT_INVALID_CODEC_SPECIFIC_OPTION;
         }
+    }
```

Comment:
Since we do not allow "end-usage" to change, we only need to call `avifProcessAOMOptionsPreInit()` before the `aom_codec_enc_init()` call. Please move lines 708-710 right before line 736, inside the `if (!codec->internal->encoderInitialized)` block.

Reply:
Done